daemon/list.go

```diff
@@ -162,10 +162,7 @@ func (daemon *Daemon) filterByNameIDMatches(view container.View, ctx *listContex
 	cntrs := make([]container.Snapshot, 0, len(matches))
 	for id := range matches {
 		c, err := view.Get(id)
-		if err != nil {
-			return nil, err
-		}
-		if c != nil {
+		if err == nil {
```

Comment:
I've updated #33882 to return a typed error. Once that is merged, I can update this one to check for it specifically.